lib/src/chrome.dart

```diff
@@ -1,3 +1,244 @@
 // Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:path/path.dart' as p;
+import 'package:webkit_inspection_protocol/webkit_inspection_protocol.dart';
+
+const _chromeEnvironment = 'CHROME_EXECUTABLE';
+const _linuxExecutable = 'google-chrome';
+const _macOSExecutable =
+    '/Applications/Google Chrome.app/Contents/MacOS/Google Chrome';
+const _windowsExecutable = r'Google\Chrome\Application\chrome.exe';
+var _windowsPrefixes = [
+  Platform.environment['LOCALAPPDATA'],
+  Platform.environment['PROGRAMFILES'],
+  Platform.environment['PROGRAMFILES(X86)']
+];
+
+String get _executable {
+  if (Platform.environment.containsKey(_chromeEnvironment)) {
+    return Platform.environment[_chromeEnvironment];
+  }
+  if (Platform.isLinux) return _linuxExecutable;
+  if (Platform.isMacOS) return _macOSExecutable;
+  if (Platform.isWindows) {
+    return p.join(
+        _windowsPrefixes.firstWhere((prefix) {
+          if (prefix == null) return false;
+          final path = p.join(prefix, _windowsExecutable);
+          return File(path).existsSync();
+        }, orElse: () => '.'),
+        _windowsExecutable);
+  }
+  throw StateError('Unexpected platform type.');
+}
+
+var _currentCompleter = Completer<Chrome>();
+
+/// A class for managing an instance of Chrome.
+class Chrome {
+  Chrome._(
+    this.debugPort,
+    this.chromeConnection, {
+    Process process,
+  }) : _process = process;
+
+  final int debugPort;
+  final Process _process;
+  final ChromeConnection chromeConnection;
+
+  /// Connects to an instance of Chrome with an open debug port.
+  static Future<Chrome> fromExisting(int port) async =>
+      _connect(Chrome._(port, ChromeConnection('localhost', port)));
+
+  static Future<Chrome> get connectedInstance => _currentCompleter.future;
+
+  /// Starts Chrome with the given arguments and a specific port.
+  ///
+  /// Each url in [urls] will be loaded in a separate tab.
+  static Future<Chrome> startWithPort(
+    List<String> urls, {
+    String userDataDir,
+    int remoteDebuggingPort,
```

Comment:
I don't think we want to be in the business of keeping track of all Chrome configuration. There are a ton of possible flags. We should probably make this more generic and just use an iterable of strings.

Reply:
Changed it to a list of strings. If we want to make some named in the future, we can do that as needed.

Reply:
If there are args that multiple clients need to know about I'd be in favor of encoding knowledge here. @grouma what is your concern? Why do you prefer to duplicate this knowledge in other clients?

Reply:
My concerns don't need to be blocking - as you mention we can always add new named args in the future.

Reply:
Nate and I discussed offline. 

My concern was having a significant list of unused optional arguments and trying to keep that in sync with the available Chrome flags. Right now we have three potential clients of this package, `package:test`, `package:webdev` and `DevTools`. From what I can tell the only arguments that are really configured /changed are `--headless` (which includes others by default) and `--remote-debugging-port`. 

We agreed that the best approach is to provide optional arguments for those two options only. We  will still pass the other options and just document their use. If and when other clients need to configure more options we can easily add another optional argument in a non-breaking way.

Ignore my suggestion of providing a mechanism to add arbitrary arguments. We can revisit that if necessary but it won't be likely given the low number of users for this package.

Sorry for the churn!